internal/runtime/runtime.go

```diff
@@ -41,7 +41,7 @@ type Runtime interface {
 	Version() string
 	SetVersion(string)
 	HooksJSONTemplate() []byte
-	PreparePackage(hooks.SDKCLIConfig, hooks.HookExecutor, types.PreparePackageOpts) error
+	PreparePackage(context.Context, hooks.SDKCLIConfig, hooks.HookExecutor, types.PreparePackageOpts) error
```

Comment:
note: Here we are updating the `package runtime` interface. Now both `InstallProjectDependencies(ctx, ...` and `PreparePackage(ctx, ...` accept context.